include/mbedtls/error.h

```diff
@@ -114,25 +114,47 @@ extern "C" {
 #define MBEDTLS_ERR_ERROR_GENERIC_ERROR       -0x0001  /**< Generic error */
 #define MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED -0x006E  /**< This is a bug in the library */
 
+/**
+ * \brief Combines a high-level and low-level error code together.
+ *
+ *        Wrapper function for mbedtls_err_add_ext(). See that function for
+ *        more details.
+ */
+#define MBEDTLS_ERROR_ADD( high, low ) \
+        mbedtls_error_add_ext( high, low, __FILE__, __LINE__ )
+
+/**
+ * \brief Testing hook called before adding/combining two error codes together.
+ *        Only used when invasive testing is enabled via MBEDTLS_TEST_HOOKS.
+ */
+extern void (*mbedtls_test_hook_error_add)( int, int, const char *, int );
 
-#if defined(MBEDTLS_TEST_HOOKS) && defined(MBEDTLS_ERROR_C)
 /**
- * \brief Set a function pointer (hook) to allow for invasive testing of error
- *        code addition.
+ * \brief Combines a high-level and low-level error code together.
  *
- * This hook is used in the test infrastructure to report on errors when
- * combining two error codes of the same level.
+ *        This function can be called directly however it is usually
+ *        called via the mbedtls_error_add macro.
  *
- * \param hook    hook to invasive testing function
+ * \note  When invasive testing is enabled via MBEDTLS_TEST_HOOKS also try to
+ *        call mbedtls_test_hook_error_add.
```

Comment:
```suggestion
 *        call \ink mbedtls_test_hook_error_add.
```
(according to https://stackoverflow.com/questions/17395098/doxygen-how-to-link-to-a-c-file-scope-global-variable, please check if it works — if you can't make it work quickly, just leave it unlinked and use `\c`)

Reply:
Done.